### ISSUE TITLE
Not enough font metrics data for `ggez::graphics::Text`.

### ISSUE THREAD

**Issue by LPeter1997** [ggez/ggez#738]
Right now it seems impossible to draw text that has a cursor in it somewhere. All because `ggez::graphics::Text` only exposes the overall text size, but there's no way to get individual character positions (so there's no way to know where to draw a cursor or selection).

There could be something like:
```
text.with_each_char(start_pos, |index, ch, pos| { /* Whatever */ });
```

**Comment by icefoxen** [comment]
Agreed, that's a good idea.  Part of the reason it doesn't do that is just 'cause nobody had asked!  Hopefully it will be pretty easy to implement something along those lines, either as an iterator to find an individual character or as a map of some kind to look it up after it's been rendered.

**Comment by IGBC** [comment]
Oh I just split the string I wanted to draw into two components then drew both with a small gap and drew a cursor inbetween

**Comment by nobbele** [comment]
should be closed with #876
